src/coprocessor/statistics/analyze.rs

```diff
@@ -330,7 +401,16 @@ impl<S: Snapshot> SampleBuilder<S> {
                 }
             }
         }
-        Ok((collectors, pk_builder))
+        let mut idx_res: Option<tipb::AnalyzeIndexResp> = None;
```

Comment:
Prefer to transform the result into protobuf in `handle_mixed` instead of in this function. We could add a struct to wrap this two result and transform it by `into_proto` in `handle_mixed`.